flags/sources.py

```diff
@@ -0,0 +1,106 @@
+
+from django.apps import apps
+from django.conf import settings
+from django.utils.module_loading import import_string
+
+from flags.conditions import get_condition
+
+
+class Condition(object):
+    """ A simple wrapper around conditions """
+
+    def __init__(self, condition, value):
+        self.condition = condition
+        self.value = value
+        self.fn = get_condition(self.condition)
+
+    def __eq__(self, other):
+        return other.condition == self.condition and other.value == self.value
+
+    def check(self, **kwargs):
+        return self.fn(self.value, **kwargs)
+
+
+class Flag(object):
+    """ A simple wrapper around feature flags and their conditions """
+
+    def __init__(self, name, conditions=[]):
+        self.name = name
+        self.conditions = conditions
+
+    def __eq__(self, other):
+        """ There can be only one feature flag of a given name """
```

Comment:
Does this also hold for tuples, or just dict values? I might update this comment if it now only applies to a specific condition.

Reply:
It just applies to feature flags themselves. The feature flag name must be unique, but it can have any number of conditions with the same name. For example, you could set a "user" condition for "william.barton" and a "user" condition for "seph.coster". Then the flag would enable for me and you, but no one else.